app/jobs/backorder_job.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class BackorderJob < ApplicationJob
+  # In the current FDC project, one shop wants to review and adjust orders
+  # before finalising. They also run a market stall and need to adjust stock
+  # levels after the market. This should be done within four hours.
+  SALE_SESSION_DELAYS = {
+    # https://openfoodnetwork.org.uk/handleyfarm/shop
+    "https://openfoodnetwork.org.uk/api/dfc/enterprises/203468" => 4.hours,
+  }.freeze
+
+  queue_as :default
+
+  def self.check_stock(order)
+    variants_needing_stock = order.variants.select do |variant|
+      # TODO: scope variants to hub.
+      # We are only supporting producer stock at the moment.
+      variant.on_hand&.negative?
+    end
+
+    linked_variants = variants_needing_stock.select do |variant|
+      variant.semantic_links.present?
+    end
+
+    return if linked_variants.empty?
+
+    # At this point we want to move to the background with perform_later.
+    # But while this is in development I'll perform the backordering
+    # immediately. It should ease debugging for now.
+    place_backorder(order, linked_variants)
+  end
+
+  def self.place_backorder(order, linked_variants)
+    user = order.distributor.owner
+
+    # We are assuming that all variants are linked to the same wholesale
+    # shop and its catalog:
```

Comment:
I'm happy with that assumption (for now 😉 ), and it will be true for the pilots we're running.